### PR TITLE
Adding NTO again to find the issue with data recollection

### PR DESCRIPTION
**Pull request by jparrill**
**What this PR does / why we need it**:
Bug hunting on data recollection after failing.
- PR Merged in the past: https://github.com/openshift/hypershift/pull/2051
- PR Reverted: https://github.com/openshift/hypershift/pull/2145

**Which issue(s) this PR fixes**:
Fixes #[HOSTEDCP-687](https://issues.redhat.com/browse/HOSTEDCP-687)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [x] This change includes unit tests.

### CODE REVIEW

**Comment by enxebre** [comment]
please squash commits before moving forward

**Comment by csrwng** [comment]
@jparrill I think not scaling down the nodes is ok, but we should also only collect journals from machines that are actually running to avoid future breakage.

File: test/e2e/nodepool_machineconfig_test.go [hunk 1]
```diff
@@ -163,13 +162,36 @@ func testNodepoolMachineconfigGetsRolledout(parentCtx context.Context, mgmtClien
 			t.Fatalf("failed to update nodepool %s after adding machineconfig: %v", nodePool.Name, err)
 		}
 
+		// DS Customization
 		ds := machineConfigUpdatedVerificationDS.DeepCopy()
+		dsName := ds.Name + "-replace"
```
Comment (csrwng) [inline test/e2e/nodepool_machineconfig_test.go:167]: nit: refactor this into a function that returns a daemonset given a daemonset name and nodepool name and call that function from here and anywhere else you need to run the daemonset.

Comment (jparrill) [reply test/e2e/nodepool_machineconfig_test.go:167]: changed, PTAL when you have some time :)

**Pull request merged**